vumi_http_proxy/http_proxy.py

```diff
@@ -102,7 +103,7 @@ def __init__(self, blacklist, resolver, http_client):
 
 class Initialize(object):
     def __init__(self, blacklistfile, ip, port):
-        blacklist = self.read_file(blacklistfile)
+        blacklist = FileReader(blacklistfile)
```

Comment:
I would pass in `blacklist` here instead of `blacklistfile` and move the file reading into `queen_of_ni` and `servicemaker` (since they can arrange to exit early without having to suddenly call `sys.exit`).